File: Chat-WebServer/AM-Chat-WebServer/Data/Requests/SendMediaMessage.cs
using AM_Chat_WebServer.Data.Models;

namespace AM_Chat_WebServer.Data.Requests;

public class SendMediaMessage
{
    public long ChannelId { get; set; }
    public string DataBase64 { get; set; }  // Zmienione z byte[] Data na string z danymi Base64
    public string Extension { get; set; }
    public MessageType MessageType { get; set; }
}

File: app/src/main/java/pwr/barwa/chat/data/requests/SendMediaMessage.kt
package pwr.barwa.chat.data.requests

import android.util.Base64
import pwr.barwa.chat.data.dto.MessageType

class SendMediaMessage(
    val channelId: Long,
    val dataBase64: String, // Teraz przesyłamy jako łańcuch Base64
    val extension: String,
    val messageType: Int
) {
    companion object {
        /**
         * Konwertuje dane binarne (ByteArray) do obiektu SendMediaMessage z danymi zakodowanymi jako Base64
         */
        fun fromByteArray(channelId: Long, byteArray: ByteArray, extension: String, messageType: MessageType): SendMediaMessage {
            val base64String = Base64.encodeToString(byteArray, Base64.NO_WRAP)
            return SendMediaMessage(channelId, base64String, extension, messageType.value)
        }
    }
}
